packages/astro/src/core/app/index.ts

```diff
@@ -335,6 +340,7 @@ export class App {
 				locals,
 				response,
 				status: response.status as 404 | 500,
+				error: response.status === 500 ? null : undefined,
```

Comment:
What's the difference between `null` and `undefined` **here**? It's a weird case. Maybe we should leave a comment that explains it

Reply:
Basically at this point of the code, we don't really have an error to report. So passing `null` means we pass nothing intentionally while `undefined` means there's no error. Hope that makes sense 😅 but ideally we can pass an error there, whatever it is

Reply:
This is valuable information! We should have this very comment in the code :)

Reply:
```suggestion
				// We don't have an error to report here. Passing null means we pass nothing intentionally
				// while undefined means there's no error
				error: response.status === 500 ? null : undefined,
```